### DISCUSSION TITLE
Modal will not close when clicking outside it

### DISCUSSION THREAD

**Discussion by nasa8x** [tailwindlabs/headlessui#599] [Ideas]
When using Modal to edit a status or content like on Facebook, but if you accidentally click out, immediately Modal will close and you have to edit from the beginning, this is really inconvenient.

This utility is quite simple but important, hope to be updated soon. 

Thank you!

**Comment by nasa8x** [comment]
https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-vue/src/components/dialog/dialog.ts#L199

`
if(outsideClick){
 useWindowEvent('mousedown', event =&gt; {      let target = event.target as HTMLElement
      if (dialogState.value !== DialogStates.Open) return      if (containers.value.size !== 1) return      if (contains(containers.value, target)) return
      api.close()      nextTick(() =&gt; target?.focus())    })
     }
`

**Comment by paulwongx** [comment]
I second this! It's a very important feature in my opinion, as so many people are using this as form inputs. @nasa8x you haven't found a way to override this behavior right?

**Reply by nasa8x** [reply]
It's here: https://github.com/tailwindlabs/headlessui/pull/613

**Reply by paulwongx** [reply]
Thanks. Alternatively someone else provided this solution
https://github.com/tailwindlabs/headlessui/issues/621
